### PR TITLE
Add text for day 19

### PR DESCRIPTION
**Pull request by mbovel**

### CODE REVIEW

**Comment by mbovel** [comment]
I took a no-bullshit approach this time, only emphasizing the important parts and not trying to explain every small step. Does this read better than day 12?

I also did not investigate performance this time, even if that would be an excellent example for runtime staging.

File: docs/2023/puzzles/day19.md [hunk 1]
```diff
@@ -6,6 +6,228 @@ import Solver from "../../../../../website/src/components/Solver.js"
...
+      case PartRegex(x, m, a, s) => Part(x.toInt, m.toInt, a.toInt, s.toInt)
+      case _                     => throw Exception(s"Invalid part: $input")
+```
+
+## Part 1 – Evaluation
+
+These helpers allow us to implement the core logic succinctly:
+
+```scala
+@main def part1: Unit = println(part1(personalPuzzle))
```
Comment (bishabosha) [inline docs/2023/puzzles/day19.md:113]:
```suggestion
def part1(input: String): Int = part1(input)
```
matching the format

File: docs/2023/puzzles/day19.md [hunk 2]
```diff
@@ -6,6 +6,228 @@ import Solver from "../../../../../website/src/components/Solver.js"
...
+    (
+      Range.safe(currentRange.from, value).map(withChannel(channel, _)),
+      Range.safe(value, currentRange.until).map(withChannel(channel, _))
+    )
+```
+
+Using these helpers, we can implement the abstract evaluator as follows:
+
+```scala
+@main def part2: Unit = println(part2(personalPuzzle, 4001))
```
Comment (bishabosha) [inline docs/2023/puzzles/day19.md:199]:
```suggestion
def part2(input: String): Long = part2(input, 4001)
```
match the format

File: docs/2023/puzzles/day19.md [hunk 3]
```diff
@@ -6,6 +6,228 @@ import Solver from "../../../../../website/src/components/Solver.js"
...
+```
+
+## Part 1 – Evaluation
+
+These helpers allow us to implement the core logic succinctly:
+
+```scala
+def part1(input: String): Int = part1(input)
+
+def part1(input: String): Int =
```
Comment (bishabosha) [inline docs/2023/puzzles/day19.md:115]:
```suggestion
def part1(input: String): Int =
```

File: docs/2023/puzzles/day19.md [hunk 4]
```diff
@@ -6,6 +6,226 @@ import Solver from "../../../../../website/src/components/Solver.js"
...
+```
+
+Using these helpers, we can implement the abstract evaluator as follows:
+
+```scala
+def part2(input: String): Long = part2(input, 4001)
+
+extension [T](part: (T, T)) private inline def swap: (T, T) = (part._2, part._1)
+
+def part2(input: String, until: Long): Long =
```
Comment (bishabosha) [inline docs/2023/puzzles/day19.md:201]:
```suggestion
def part2(input: String): Long = combinations(input, 4001)

extension [T](part: (T, T)) private inline def swap: (T, T) = (part._2, part._1)

def combinations(input: String, until: Long): Long =
```

**Review by bishabosha — APPROVED** [review]
its a great demonstration of data modelling, thanks!

**Pull request merged**